crates/breez-sdk/core/src/sdk_builder.rs

```diff
@@ -265,16 +284,38 @@ impl SdkBuilder {
         };
         let shutdown_sender = watch::channel::<()>(()).0;
 
+        let event_emitter = Arc::new(EventEmitter::new());
+        let storage = if let Some(server_url) = &self.real_time_sync_server_url {
+            let Some(sync_storage) = self.sync_storage else {
+                return Err(SdkError::Generic("Sync storage is not available".into()));
+            };
+
+            init_and_start_real_time_sync(RealTimeSyncParams {
+                server_url: server_url.clone(),
+                api_key: self.config.api_key.clone(),
```

Comment:
Just want to check what happens when no API key is provided (e.g the Breez itests run in regtest with no API key), does it gracefully handle failed calls to the data sync server?